backend/modules/publishing/router.py

```
@@ -1 +1,444 @@
+        logger.error(f"Error in update_simulation_status: {e}", exc_info=True)
+        db.rollback()
+        raise HTTPException(
+            status_code=500,
+            detail=f"Failed to update simulation status: {str(e)}"
+        )
+
+
+@router.delete("/{simulation_id}", status_code=204)
+async def delete_simulation(
+    simulation_id: int,
+    db: Session = Depends(get_db),
```

Ruff (0.14.8)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)